core/commands/show_command.rb

```diff
@@ -81,7 +81,10 @@ def execute
 
   def show_private_ip_address(params)
     Configuration.from_spec(params.first).and_then do |config|
-      network_settings = NetworkSettings.from_file(config.network_settings_file)
+      result = NetworkSettings.from_file(config.network_settings_file)
+      return result if result.error?
```

Comment:
Instead you must use `and_then` method.

In current architecture you can do the following:
```ruby
Configuration.from_spec(...).and_then do |config|
  NetworkSettings.from_file(...)
end.and_then do |network_settings|
  ...
end
```
Please consult me when you will try to implement this change.